src/dnaio/_core.pyx

```diff
@@ -146,6 +150,27 @@ cdef class Sequence:
         return self.fastq_bytes(two_headers=True)
 
 
+cdef class SequenceBytes(Sequence):
```

Comment:
I don’t think deriving from `Sequence` is the proper thing to do. For example, `qualities_as_bytes` won’t work because it calls `self.qualities.encode()`. `__repr__` is also wrong at the moment. Also, since `Sequence.__cinit__` now gets `object` arguments, the type checking done by Cython is lost. Some of this is fixable, but then the main issue remains that `SequenceBytes` cannot be used wherever a `Sequence` is expected, that is, the Liskov Substitution Principle is violated.

Reply:
Agreed. I will do a simple copypaste then with some minor alterations.

Sequence and BytesSequence can share a lot, but Cython makes that very hard to do. It can be implemented more easily directly in the Python C-API. But I will do that in a later PR.